core/unit_test/TestDefaultDeviceTypeInit.hpp

```diff
@@ -293,8 +293,13 @@ void check_correct_initialization(const Kokkos::InitArguments& argstruct) {
 
 // TODO: Add check whether correct number of threads are actually started.
 void test_no_arguments() {
+  Kokkos::finalize();
+  Kokkos::finalize();
   Kokkos::initialize();
   check_correct_initialization(Kokkos::InitArguments());
+  Kokkos::initialize();
+  check_correct_initialization(Kokkos::InitArguments());
+  Kokkos::finalize();
   Kokkos::finalize();
 }
```

Comment:
I think this should be a separate test case.

Reply:
Do we also have a test for initializing individual execution space types as `Legion` requires?
To me, it still feels a little weird to allow calling `finalize` before `initialize (but that seems to preserve previous behavior).

Reply:
I too would like to discuss these changes